src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -716,16 +716,12 @@ module T = struct
           let t =
             d.transaction_with_info |> Ledger.Transaction_applied.transaction
           in
-          t :: acc)
+          (t, d.statement) :: acc)
```

Comment:
This whole fold is unnecessary, the fold below could use `data` directly and the `Error` case below could extract the `txn`s from `data` there.